packages/block-editor/src/hooks/use-bindings-attributes.js

```diff
@@ -22,87 +25,224 @@ import { unlock } from '../lock-unlock';
+ * This component is responsible for detecting and
+ * propagating data changes from the source to the block.
+ *
+ * @param {Object} props            - The component props.
+ * @param {string} props.attrName   - The attribute name.
+ * @param {any}    props.attrValue  - The attribute value.
+ * @param {string} props.blockName  - The block name.
+ * @param {Object} props.blockProps - The block props with bound attribute.
+ * @param {Object} props.source     - Source handler.
+ * @param {Object} props.args       - The arguments to pass to the source.
+ * @return {null}                     This is a data-handling component. Render nothing.
+ */
+const BindingConnector = ( {
+	args,
+	attrName,
+	attrValue,
+	blockName,
+	blockProps,
+	source,
+} ) => {
+	const { placeholder, value: propValue } = source.useSource(
+		blockProps,
+		args
+	);
+
+	const setAttributes = blockProps.setAttributes;
+
+	const { syncDerivedUpdates } = unlock( useDispatch( blockEditorStore ) );
+
+	const updateBoundAttibute = useCallback(
+		( newAttrValue, prevAttrValue ) => {
+			/*
+			 * If the attribute is a RichTextData instance,
+			 * (core/paragraph, core/heading, core/button, etc.)
+			 * compare its HTML representation with the new value.
+			 *
+			 * To do: it looks like a workaround.
+			 * Consider improving the attribute and metadata fields types.
+			 */
+			if ( prevAttrValue instanceof RichTextData ) {
+				// Bail early if the Rich Text value is the same.
+				if ( prevAttrValue.toHTMLString() === newAttrValue ) {
+					return;
+				}
+
+				/*
+				 * To preserve the value type,
+				 * convert the new value to a RichTextData instance.
+				 */
+				newAttrValue = RichTextData.fromHTMLString( newAttrValue );
 			}
 
-			return (
-				<BlockEdit
-					key="edit"
-					{ ...props }
-					attributes={ updatedAttributes }
-				/>
-			);
+			if ( prevAttrValue === newAttrValue ) {
+				return;
+			}
+
+			syncDerivedUpdates( () => {
+				setAttributes( {
+					[ attrName ]: newAttrValue,
+				} );
+			} );
 		},
-		'useBoundAttributes'
+		[ attrName, setAttributes, syncDerivedUpdates ]
+	);
+
+	useEffect( () => {
+		if ( typeof propValue !== 'undefined' ) {
+			updateBoundAttibute( propValue, attrValue );
+		} else if ( placeholder ) {
+			/*
+			 * Placeholder fallback.
+			 * If the attribute is `src` or `href`,
+			 * a placeholder can't be used because it is not a valid url.
+			 * Adding this workaround until
+			 * attributes and metadata fields types are improved and include `url`.
+			 */
+			const htmlAttribute =
+				getBlockType( blockName ).attributes[ attrName ].attribute;
+
+			if ( htmlAttribute === 'src' || htmlAttribute === 'href' ) {
+				updateBoundAttibute( null );
+				return;
+			}
+
+			updateBoundAttibute( placeholder );
+		}
+	}, [
+		updateBoundAttibute,
+		propValue,
+		attrValue,
+		placeholder,
+		blockName,
+		attrName,
+	] );
+
+	return null;
+};
+
+function BlockBindingBridge( { bindings, props, blockName, attributes } ) {
```

Comment:
`props.props` might be hard to follow for folks. How about using more descriptive name here?